// plugin-welcome/src/api/models/EntDepartmentEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Doctor
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntDoctor,
    EntDoctorFromJSON,
    EntDoctorFromJSONTyped,
    EntDoctorToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntDepartmentEdges
 */
export interface EntDepartmentEdges {
    /**
     * Doctor holds the value of the doctor edge.
     * @type {Array<EntDoctor>}
     * @memberof EntDepartmentEdges
     */
    doctor?: Array<EntDoctor>;
}

export function EntDepartmentEdgesFromJSON(json: any): EntDepartmentEdges {
    return EntDepartmentEdgesFromJSONTyped(json, false);
}

export function EntDepartmentEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntDepartmentEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'doctor': !exists(json, 'doctor') ? undefined : ((json['doctor'] as Array<any>).map(EntDoctorFromJSON)),
    };
}

export function EntDepartmentEdgesToJSON(value?: EntDepartmentEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'doctor': value.doctor === undefined ? undefined : ((value.doctor as Array<any>).map(EntDoctorToJSON)),
    };
}
