// plugin-welcome/src/api/models/EntDoctorEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Doctor
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntDegree,
    EntDegreeFromJSON,
    EntDegreeFromJSONTyped,
    EntDegreeToJSON,
    EntDepartment,
    EntDepartmentFromJSON,
    EntDepartmentFromJSONTyped,
    EntDepartmentToJSON,
    EntNametitle,
    EntNametitleFromJSON,
    EntNametitleFromJSONTyped,
    EntNametitleToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntDoctorEdges
 */
export interface EntDoctorEdges {
    /**
     * 
     * @type {EntDegree}
     * @memberof EntDoctorEdges
     */
    degree?: EntDegree;
    /**
     * 
     * @type {EntDepartment}
     * @memberof EntDoctorEdges
     */
    department?: EntDepartment;
    /**
     * 
     * @type {EntNametitle}
     * @memberof EntDoctorEdges
     */
    nametitle?: EntNametitle;
}

export function EntDoctorEdgesFromJSON(json: any): EntDoctorEdges {
    return EntDoctorEdgesFromJSONTyped(json, false);
}

export function EntDoctorEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntDoctorEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'degree': !exists(json, 'Degree') ? undefined : EntDegreeFromJSON(json['Degree']),
        'department': !exists(json, 'Department') ? undefined : EntDepartmentFromJSON(json['Department']),
        'nametitle': !exists(json, 'Nametitle') ? undefined : EntNametitleFromJSON(json['Nametitle']),
    };
}

export function EntDoctorEdgesToJSON(value?: EntDoctorEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'degree': EntDegreeToJSON(value.degree),
        'department': EntDepartmentToJSON(value.department),
        'nametitle': EntNametitleToJSON(value.nametitle),
    };
}
